modules/web/src/plugins/ballerina/diagram/views/default/components/nodes/try-statement-decorator.jsx

```diff
@@ -0,0 +1,444 @@
+
+        if (model.catchBlocks.length === 0 && finallyStmt) {
+            p5Y += titleH / 2;
+        }
+
+        const p7X = p1X;
+        const p7Y = p5Y;
+
+        const p8X = bBox.x;
+        const p8Y = p2Y + (titleH / 2);
+
+        const p11X = p1X;
+        const p11Y = p1Y + (titleH / 2);
+
+        const actionBoxBbox = new SimpleBBox();
+        actionBoxBbox.w = (3 * designer.config.actionBox.width) / 4;
+        actionBoxBbox.h = designer.config.actionBox.height;
+        actionBoxBbox.x = p8X - (actionBoxBbox.w / 2);
+        actionBoxBbox.y = p8Y;
+
+        let statementRectClass = 'statement-title-rect';
+        if (isDebugHit) {
+            statementRectClass = `${statementRectClass} debug-hit`;
+        }
+
+        const body = getComponentForNodeArray(this.props.model.body);
+        const disableDeleteForFinally = model.catchBlocks.length <= 0 && model.finallyBody;
+        const disableDeleteForCatch = model.catchBlocks.length === 1 && (!model.finallyBody);
+
+        return (
+            <g
+                onMouseOut={this.setActionVisibilityFalse}
+                onMouseOver={this.setActionVisibilityTrue}
+                ref={(group) => {
+                    this.myRoot = group;
+                }}
+            >
+                <polyline
+                    points={`${p3X},${p3Y} ${p4X},${p4Y} ${p5X},${p5Y} ${p7X},${p7Y} ${p11X},${p11Y} ${p2X},${p2Y}`}
+                    className='background-empty-rect'
+                />
+                <rect
+                    x={p2X}
+                    y={p1Y}
+                    width={titleW}
+                    height={titleH}
+                    className={statementRectClass}
+                    rx='5'
+                    ry='5'
+                />
+                <text
+                    x={p8X}
+                    y={p2Y}
+                    className='statement-title-text'
+                >try
+                </text>
+                <DropZone
+                    x={p11X}
+                    y={p11Y}
+                    width={statementBBox.w}
+                    height={statementBBox.h}
+                    baseComponent='rect'
+                    dropTarget={this.props.model.body}
+                    enableDragBg
+                    enableCenterOverlayLine={!this.props.disableDropzoneMiddleLineOverlay}
+                />
+                { isBreakpoint && this.renderBreakpointIndicator() }
+                {this.props.children}
+                {body}
+                <ActionBox
+                    bBox={actionBoxBbox}
+                    show={this.state.active}
+                    isBreakpoint={isBreakpoint}
+                    onDelete={() => this.onDelete()}
+                    onJumptoCodeLine={() => this.onJumpToCodeLine()}
+                    onBreakpointClick={() => this.props.onBreakpointClick()}
+                    disableButtons={this.props.disableButtons}
+                />
+                {(() => {
```

Comment:
This render function is too long and difficult to read? shall we break this into several functions?

Reply:
This includes some inter-dependent code on drawing components related to catch in try catch. i.e. drawing connector lines to finally block and expanding the vertical lifeline from catch up to the 'finally' block which requires manipulating the coordinates. Therefore, it is not feasible to move this logic to a different function.